pallets/loans/src/lib.rs

```diff
@@ -659,8 +651,37 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
             Self::ensure_currency(&currency_id)?;
+            ensure!(
+                AccountDeposits::<T>::contains_key(currency_id, &who),
+                Error::<T>::NoDeposit
+            );
+            let mut deposits = Self::account_deposits(currency_id, &who);
+            if deposits.is_collateral == enable {
+                return Err(Error::<T>::DuplicateOperation.into());
+            }
+            // turn on the collateral button
+            if enable {
+                deposits.is_collateral = true;
+                AccountDeposits::<T>::insert(currency_id, &who, deposits);
+                Self::deposit_event(Event::<T>::CollateralAssetAdded(who, currency_id));
+
+                return Ok(().into());
+            }
+            // turn off the collateral button after checking the liquidity
+            let total_collateral_asset_value = Self::total_collateral_asset_value(&who)?;
+            let collateral_asset_value = Self::collateral_asset_value(&who, &currency_id)?;
+            let total_borrowed_value = Self::total_borrowed_value(&who)?;
+            if total_collateral_asset_value
+                < total_borrowed_value
```

Comment:
we need to check if ` total_borrowed_value.is_zero`, if yes then allow users to disable

Reply:
It already covers the case. You can check this test https://github.com/parallel-finance/parallel/pull/206/files#diff-dc5bf2eed1f7171ad846871fe225e6434f7ffcba1303f00668394d0bcda8b190R331